live-build/misc/migration-scripts/dx_execute

```diff
@@ -51,6 +51,23 @@ EOM
 	exit 2
 }
 
+function upgrade_alert() {
+	local pid
+
+	#
+	# This could hang if the JVM is in a bad state, so we spawn it in the
+	# background and give it 5 seconds to complete before giving up.
+	#
+	/opt/delphix/server/bin/jmxtool boot upgrade server &>/dev/null &
```

Comment:
Can we just use `timeout` instead? It would simplify this logic.

Reply:
Yeah, much simpler indeed.